src/compiler/transpile/create-build-program.ts

```diff
@@ -1,67 +1,124 @@
 import type * as d from '../../declarations';
 import { getTsOptionsToExtend } from './ts-config';
-import { GENERATED_DTS } from '../output-targets/output-utils';
 import ts from 'typescript';
 
+/**
+ * Create a TypeScript Program ({@link ts.Program}) to perform builds of a Stencil project using the provided
+ * `buildCallback` entity
+ * @param config a Stencil configuration to apply to a full build of a Stencil project
+ * @param buildCallback a callback that invokes the actual transpilation of a Stencil project
+ * @returns a Program that marries the TypeScript and Stencil compilers together.
+ */
 export const createTsBuildProgram = async (
   config: d.Config,
   buildCallback: (tsBuilder: ts.BuilderProgram) => Promise<void>
-) => {
-  let isRunning = false;
-  let timeoutId: any;
+): Promise<ts.WatchOfConfigFile<ts.EmitAndSemanticDiagnosticsBuilderProgram>> => {
+  let isBuildRunning = false;
+  let currentBuildTimeoutId: any;
 
   const optionsToExtend = getTsOptionsToExtend(config);
 
+  /**
+   * Create a {@link ts.System}. The System is responsible for handling all interactions between the TypeScript compiler
+   * and the host operating system.
+   */
   const tsWatchSys: ts.System = {
     ...ts.sys,
 
-    watchFile(path, callback) {
-      if (path.endsWith(`/${GENERATED_DTS}`)) {
-        return ts.sys.watchFile(path, callback);
-      }
```

Comment:
Yup! That's it.

So admission time - I'm not entirely sure if/why `close` was overridden the way it was.  I followed the pattern to avoid breaking anything further (since this isn't under test ATM). Unfortunately there isn't a design doc or anything in `git blame` that makes the justification for this immediately obvious 😢

Reply:
hmmm the mystery deepends...in that case I think it's fine as is 😄